builder/vsphere/driver/vm.go

```diff
@@ -1096,29 +1101,39 @@ func findNetwork(network string, host string, d *VCenterDriver) (object.NetworkR
 	return nil, fmt.Errorf("error finding network; 'host' and 'network' not specified. at least one of the two must be specified")
 }
 
-func newVirtualPCIPassthroughAllowedDevice(devices []PCIPassthroughAllowedDevice) types.VirtualPCIPassthrough {
+func newVirtualPCIPassthroughAllowedDevice(devices []PCIPassthroughAllowedDevice) (types.VirtualPCIPassthrough, error) {
 	allowedDevices := make([]types.VirtualPCIPassthroughAllowedDevice, len(devices))
 	for i, device := range devices {
 		deviceId, err := strconv.ParseInt(device.DeviceId, 16, 32)
 		if err != nil {
-			// handle error, for example:
-			log.Printf("Error parsing DeviceId: %v\n", err)
-			continue
+			return types.VirtualPCIPassthrough{}, fmt.Errorf("error parsing DeviceId %s: %v", device.DeviceId, err)
 		}
+		if deviceId > math.MaxInt32 || deviceId < math.MinInt32 {
+			return types.VirtualPCIPassthrough{}, fmt.Errorf("DeviceId %s exceeds int32 range", device.DeviceId)
+		}
+
 		vendorId, err := strconv.ParseUint(device.VendorId, 16, 32)
```

Comment:
Is there a reason why this is a `ParseUint` and not a `ParseInt`? Or the other way around, I would think that `deviceId` cannot be negative, can it?